app/workers/preneeds/delete_old_uploads.rb

```diff
@@ -1,30 +1,17 @@
 # frozen_string_literal: true
 
 module Preneeds
-  class DeleteOldUploads
-    include Sidekiq::Worker
+  class DeleteOldUploads < DeleteAttachmentJob
+    ATTACHMENT_CLASSES = %w[Preneeds::PreneedAttachment].freeze
+    FORM_ID = '40-10007'
 
-    sidekiq_options(unique_for: 30.minutes, retry: false)
-
-    EXPIRATION_TIME = 2.months
-
-    def perform
-      Sentry::TagRainbows.tag
-
-      Preneeds::PreneedAttachment.where(
-        'created_at < ?', EXPIRATION_TIME.ago
-      ).where.not(guid: uuids_to_keep).find_each(&:destroy!)
-    end
-
-    def uuids_to_keep
+    def get_uuids(form_data)
       uuids = []
 
-      InProgressForm.where(form_id: '40-10007').find_each do |in_progress_form|
-        attachments = in_progress_form.data_and_metadata[:form_data]['preneedAttachments']
-        if attachments.present?
-          attachments.each do |attachment|
-            uuids << attachment['confirmationCode']
-          end
+      attachments = form_data['preneedAttachments']
+      if attachments.present?
+        attachments.each do |attachment|
```

Comment:
Nitpicky:  I like your use of `Array.wrap` in vic/delete_old_uploads.  We can change line 11 to `attachments = Array.wrap(form_data['preneedAttachments'])` and then remove the conditional on line 12.